internal/yggdrasil/yggdrasil.go

```diff
@@ -141,6 +162,9 @@ func (h *Handler) GetControlMessageForDevice(ctx context.Context, params yggdras
 
 func (h *Handler) GetDataMessageForDevice(ctx context.Context, params yggdrasil.GetDataMessageForDeviceParams) middleware.Responder {
 	deviceID := params.DeviceID
+	if !IsOwnDevice(ctx, deviceID) {
```

Comment:
Not sure if the generated code can allow it, but I don't like the fact we need to repeat the check in any call. Not that we plan on extending the APIs, but I'd prefer kind of a "template" function that authorization logic is located in a single place and all the calls have to go through it.
But not sure how the generated code can be customized to allow it.

Reply:
We had an offline conversation here, it's hard to get the params.deviceID in a middleware call, and not in the autogenerated code too,  I cannot see a better way here :-( (I do not really liked too)